### ISSUE TITLE
[MapperParsingException[failed to parse]; nested: IllegalStateException[Mixing up field types: class org.elasticsearch.index.mapper.core.StringFieldMapper$StringFieldType != class org.elasticsearch.index.mapper.geo.BaseGeoPointFieldMapper$GeoPointFieldType on field coordinates

### ISSUE THREAD

**Issue by manaligaikwad** [elastic/elasticsearch#18667]
![es_exception](https://cloud.githubusercontent.com/assets/12480500/15691513/7d1f6aa8-2746-11e6-970f-c2f976919a16.PNG)

**Elasticsearch version**: 2.3.3

**JVM version**:1.8

**OS version**: Windows 10

**Description of the problem including expected versus actual behavior**:
I am trying to add mapping as a geo_points to the field named "coordinates.". Index is getting created with this mapping, but after that when I am using java bulk index api. While adding type to index, I am getting following exception.
message [MapperParsingException[failed to parse]; nested: IllegalStateException[Mixing up field types: class org.elasticsearch.index.mapper.core.StringFieldMapper$StringFieldType != class org.elasticsearch.index.mapper.geo.BaseGeoPointFieldMapper$GeoPointFieldType on field coordinates];]

This is the mapping I want to add - 
{
    "properties": {
     "coordinates": {
        "type": "geo_point"
      }
    }
}

When I checked mapping for an index and it's type it is different -
{
  "testgeo": { //index
    "mappings": {
      "twitter_mapping": {
        "properties": {
          "coordinates": {
            "type": "geo_point"
          }
        }
      },
      "test": {} //index type
    }
  }
}

I tried to create mapping with different types (other than geo_point), my code works fine with other types. Like this -
{
  "tweet_media": {    //index
    "mappings": {
      "test": {      //index type
        "properties": {
          "created_at": {
            "type": "date",
            "format": "YYYY-MM-DD'T'HH:mm:ss.SSSZ"
          },
          "expanded_url": {
            "type": "string",
            "analyzer": "case_insensitive_keyword_analyzer"
          },
          "has_media": {
            "type": "boolean"
          },
          "hashtags": {
            "type": "string",
            "analyzer": "tag_analyzer"
          },
          "language": {
            "type": "string",
            "analyzer": "en_analyzer"
          },
          "name": {
            "type": "string",
            "analyzer": "case_insensitive_keyword_analyzer"
          },
          "text": {
            "type": "string",
            "analyzer": "en_analyzer"
          },
          "user_mentions": {
            "type": "string",
            "analyzer": "case_insensitive_keyword_analyzer"
          }
        }
      },
      "twitter_mapping": {
        "properties": {
          "created_at": {
            "type": "date",
            "format": "YYYY-MM-DD'T'HH:mm:ss.SSSZ"
          },
          "expanded_url": {
            "type": "string",
            "analyzer": "case_insensitive_keyword_analyzer"
          },
          "has_media": {
            "type": "boolean"
          },
          "hashtags": {
            "type": "string",
            "analyzer": "tag_analyzer"
          },
          "language": {
            "type": "string",
            "analyzer": "en_analyzer"
          },
          "name": {
            "type": "string",
            "analyzer": "case_insensitive_keyword_analyzer"
          },
          "text": {
            "type": "string",
            "analyzer": "en_analyzer"
          },
          "user_mentions": {
            "type": "string",
            "analyzer": "case_insensitive_keyword_analyzer"
          }
        }
      }
    }
  }
}

**Steps to reproduce**:
 1.
 2.
 3.

**Provide logs (if relevant)**:
failure in bulk execution:
[1]: index [testgeo], type [test], id [AVUIFTEIEfgnJAE-P1J7], message [MapperParsingException[failed to parse]; nested: IllegalStateException[Mixing up field types: class org.elasticsearch.index.mapper.core.StringFieldMapper$StringFieldType != class org.elasticsearch.index.mapper.geo.BaseGeoPointFieldMapper$GeoPointFieldType on field coordinates];]

**Describe the feature**:

**Comment by clintongormley** [comment]
You're trying to index a field called `location` with a string value (which is not a valid geo-point) into a field called `location` which is mapped as a `geo_point`.

**Comment by Gramler** [comment]
I ran into this problem when my mapping type I used to create the mapping (map1) and the mapping type I passed when writing to the index (map2) was different. It then creates a new mapping on the fly for the "map2" that is a String, or Double, depending on the type of geoPoint info I had in there. Anyway... it then falls over with the given error message when it tries to write to the index

So double check your mapping types. It might be that.
